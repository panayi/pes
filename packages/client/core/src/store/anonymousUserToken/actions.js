import { createAction } from 'redux-actions';
import * as types from './types';

export const reset = createAction(types.RESET);
