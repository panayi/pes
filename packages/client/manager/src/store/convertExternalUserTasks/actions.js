import * as convertExternalUserTasks from '@pesposa/core/src/client/convertExternalUserTasks';

export const createEngagement = (externalUserId, channel, content) => (
  dispatch,
  getState,
  getFirebase,
) => {
  const engagement = {
    channel,
    content,
  };

  return convertExternalUserTasks.createEngagement(
    getFirebase(),
    externalUserId,
    engagement,
  );
};

export const archive = externalUserId => (dispatch, getState, getFirebase) =>
  convertExternalUserTasks.archive(getFirebase(), externalUserId);

export const sendEmail = (externalUserId, adId) => (
  dispatch,
  getState,
  getFirebase,
) => convertExternalUserTasks.sendEmail(getFirebase(), externalUserId, adId);
