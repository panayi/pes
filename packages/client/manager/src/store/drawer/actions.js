import { createAction } from 'redux-actions';
import * as types from './types';
import * as selectors from './selectors';

export const open = createAction(types.OPEN);

export const close = createAction(types.CLOSE);

export const toggle = () => (dispatch, getState) => {
  const isOpened = selectors.isOpenedSelector(getState());

  return isOpened ? dispatch(close()) : dispatch(open());
};
