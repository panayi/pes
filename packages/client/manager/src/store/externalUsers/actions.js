import * as externalUsers from '@pesposa/core/src/client/externalUsers';

export const create = externalUser => (dispatch, getState, getFirebase) =>
  externalUsers.create(getFirebase(), externalUser);

export const update = (id, externalUser) => (dispatch, getState, getFirebase) =>
  externalUsers.update(getFirebase(), id, externalUser);

export const remove = id => (dispatch, getState, getFirebase) =>
  externalUsers.remove(getFirebase(), id);
