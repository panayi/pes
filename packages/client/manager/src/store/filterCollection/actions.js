import { createAction } from 'redux-actions';
import zipArgs from '@pesposa/core/src/utils/zipArgs';
import * as types from './types';

export const setSelected = createAction(
  types.SET_SELECTED,
  zipArgs(['id', 'value']),
);

export const setQuery = createAction(types.SET_QUERY, zipArgs(['id', 'value']));
