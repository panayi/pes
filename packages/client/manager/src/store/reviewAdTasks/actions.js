import * as reviewAdTasks from '@pesposa/core/src/client/reviewAdTasks';

export const approve = id => async (dispatch, getState, getFirebase) =>
  reviewAdTasks.approve(getFirebase(), id);

export const reject = (id, rejectionReason) => (
  dispatch,
  getState,
  getFirebase,
) => reviewAdTasks.reject(getFirebase(), id, rejectionReason);
