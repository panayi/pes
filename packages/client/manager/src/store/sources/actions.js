import * as sources from '@pesposa/core/src/client/sources';

export const create = source => (dispatch, getState, getFirebase) =>
  sources.create(getFirebase(), source);

export const update = (id, source) => (dispatch, getState, getFirebase) =>
  sources.update(getFirebase(), id, source);

export const remove = id => (dispatch, getState, getFirebase) =>
  sources.remove(getFirebase(), id);
