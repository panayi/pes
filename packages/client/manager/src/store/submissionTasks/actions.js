import * as R from 'ramda';
import { selectors as authSelectors } from '@pesposa/client-core/src/store/firebase/auth';
import * as submissionTasks from '@pesposa/core/src/client/submissionTasks';
import { models } from '@pesposa/client-core/src/store/firebase/data';

export const create = submission => (dispatch, getState, getFirebase) => {
  const finalSubmission = R.assoc(
    'submitter',
    authSelectors.uidSelector(getState()),
    submission,
  );
  return submissionTasks.create(getFirebase(), finalSubmission);
};

export const updateSubmission = (id, submission) => (
  dispatch,
  getState,
  getFirebase,
) => {
  const finalSubmission = R.filter(R.identity, submission);
  return submissionTasks.updateSubmission(getFirebase(), id, finalSubmission);
};

export const accept = (id, submission) => async (
  dispatch,
  getState,
  getFirebase,
) => {
  const { location } = submission;
  const locations = models.locations.all.selector(getState());
  const locationObject = R.compose(
    R.omit(['id']),
    R.find(R.propEq('id', location)),
  )(locations);
  const adProps = R.compose(
    R.filter(R.identity),
    R.assoc('location', locationObject),
    R.pick(['title', 'body', 'price', 'images', 'category']),
  )(submission);

  return submissionTasks.accept(getFirebase(), id, adProps);
};

export const reject = id => (dispatch, getState, getFirebase) =>
  submissionTasks.reject(getFirebase(), id);
