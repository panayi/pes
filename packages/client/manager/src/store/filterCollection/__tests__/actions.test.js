import * as actions from '../actions';

describe('filterCollection', () => {
  describe('actions', () => {
    describe('setQuery', () => {
      it('should create SET_QUERY action with correct payload', () => {
        expect(actions.setQuery('foo', 'bar')).toMatchSnapshot();
      });
    });
  });
});
