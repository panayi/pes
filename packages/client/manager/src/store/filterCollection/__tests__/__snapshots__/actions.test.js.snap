// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterCollection actions setQuery should create SET_QUERY action with correct payload 1`] = `
Object {
  "payload": Object {
    "id": "foo",
    "value": "bar",
  },
  "type": "filterCollection/setQuery",
}
`;
